src/pybind/mgr/cephadm/module.py

```diff
@@ -2099,14 +2139,24 @@ def add_mgr(self, spec):
         return self._add_daemon('mgr', spec, self._create_mgr)
 
     def apply_mgr(self, spec):
+        self.save_spec(spec)
+        self._kick_serve_loop()
+        return trivial_result("scheduled MGR creation..")
+
+    def _apply_mgr(self, spec):
         # type: (orchestrator.ServiceSpec) -> AsyncCompletion
         return self._apply_service('mgr', spec, self._create_mgr)
 
     def add_mds(self, spec):
         # type: (orchestrator.ServiceSpec) -> AsyncCompletion
         return self._add_daemon('mds', spec, self._create_mds, self._config_mds)
 
-    def apply_mds(self, spec):
+    def apply_mds(self, spec: orchestrator.ServiceSpec) -> orchestrator.Completion:
+        self.save_spec(spec)
+        self._kick_serve_loop()
+        return trivial_result("scheduled MDS creation..")
```

Comment:
search/replace "scheduled __ creation.." with "Scheduled __ creation..."?

Reply:
or, make save_spec() return ``trivial_result('Scheduled %s creation...' % service_name))`` ?

Reply:
Maybe call it ``apply_spec`` and have it call _kick_serve_loop and return the trivial_result...

Reply:
Agree on the renaming.

re: https://github.com/ceph/ceph/pull/33553#discussion_r385333518

we also call `save_spec` in `apply_service_config` method which doesn't really _schedule_ a creation but really just saves it. Unless we enhance `save_spec` and `apply_service_config` to handle those events, it's going to be confusing. I'd vote for keeping it the way it is.